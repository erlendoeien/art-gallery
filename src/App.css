.app {
    overflow: hidden;
    min-height: 100vh;
}

::-webkit-scrollbar {
    width: 0.5vw;
}

::-webkit-scrollbar-thumb {
    background: #111;
    border-radius: 10px;
}

::-webkit-scrollbar-track {
    background: #eee;
    border-radius: 10px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
    background: #555;
}
