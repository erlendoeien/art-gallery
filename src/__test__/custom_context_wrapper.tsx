/* eslint react/prop-types: 0 */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '../components/ThemeProvider';
import { MoodProvider } from '../components/MoodProvider';

// Custom wrapper for render function,
// which wraps the component inside ThemeProvider and MoodProvider
export const Wrapper: React.ComponentType = ({ children }) => {
    return (
        <ThemeProvider>
            <MoodProvider>
                <MemoryRouter>{children}</MemoryRouter>
            </MoodProvider>
        </ThemeProvider>
    );
};

// re-export everything
export * from '@testing-library/react';
