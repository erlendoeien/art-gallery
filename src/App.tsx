// eslint-disable-next-line
import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import GalleryController from './components/Gallery/GalleryController';
import LandingPage from './components/LandingPage';
import Navbar from './components/Navbar';
import './App.css';
import { useAppSelector } from './app/hooks';

const App: React.FC = () => {
    const { fontColor, backgroundColor } = useAppSelector((state) => state.theme);
    return (
        <Router>
            <div className="app" style={{ backgroundColor, color: fontColor }}>
                <Navbar />
                <Switch>
                    <Route exact path="/">
                        <LandingPage />
                    </Route>
                    <Route path="/Utstilling">
                        <GalleryController displayFavorites={false} />
                    </Route>
                    <Route path="/favoritter">
                        <GalleryController displayFavorites />
                    </Route>
                </Switch>
            </div>
        </Router>
    );
};

export default App;
