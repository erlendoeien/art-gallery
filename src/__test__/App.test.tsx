import React from 'react';
import { render, Wrapper } from './custom_context_wrapper';
import App from '../App';

it('renders correctly', () => {
    const { container } = render(<App />, {
        wrapper: Wrapper
    });
    // snapshot testing
    expect(container).toMatchSnapshot();
});

test('renders title', () => {
    const { getByText } = render(<App />);
    const headerElement = getByText(/Velkommen til vektorgalleriet!/i);
    expect(headerElement).toBeInTheDocument();
});
