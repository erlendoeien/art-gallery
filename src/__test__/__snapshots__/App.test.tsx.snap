// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="app"
    style="background-color: rgb(255, 255, 255); color: rgb(30, 30, 30);"
  >
    <div
      class="navbar"
      style="background-color: rgb(238, 238, 238); box-shadow: -1px 0px 15px 0px #A6A6A6; height: 30px;"
    >
      <div
        class=""
        id="navIcon"
        role="button"
        tabindex="0"
      >
        <span
          style="background: rgb(30, 30, 30);"
        />
        <span
          style="background: rgb(30, 30, 30);"
        />
        <span
          style="background: rgb(30, 30, 30);"
        />
      </div>
      <div
        class="settingsBtn"
      >
        <input
          alt="instillinger-knapp"
          height="30px"
          src="settings.png"
          type="image"
        />
      </div>
      <div
        class="noSettings"
        style="background-color: rgb(255, 255, 255);"
      >
        <div>
          <h2
            style="color: rgb(30, 30, 30);"
          >
            Tilpass siden
          </h2>
          <h3
            style="color: rgb(30, 30, 30);"
          >
            Velg mellom dark mode eller light mode:
          </h3>
          <div
            class="flexWrapper"
          >
            <label
              class="switch"
              for="darkmode"
            >
              <input
                id="darkmode"
                type="checkbox"
              />
              <span
                class="slider"
              />
            </label>
            <img
              alt="Sol eller måne"
              class="themeIcon"
              src="sun.png"
            />
          </div>
          <h3
            style="color: rgb(30, 30, 30);"
          >
             
            Hvilket humør er du i dag? 
            <br />
            Velg mellom glade eller triste kunstverk:
          </h3>
          <div
            class="flexWrapper"
          >
            <label
              class="switch"
              for="mood"
            >
              <input
                checked=""
                id="mood"
                type="checkbox"
              />
              <span
                class="slider"
              />
            </label>
            <img
              alt="Glad eller Trist"
              class="themeIcon"
              src="sad.png"
            />
          </div>
        </div>
      </div>
      <ul
        class="navLinks "
      >
        <li>
          <a
            aria-current="page"
            class="nav_link activeNavLink"
            href="/"
            style="color: rgb(30, 30, 30);"
          >
            Forside
          </a>
        </li>
        <li>
          <a
            class="nav_link"
            href="/utstilling"
            style="color: rgb(30, 30, 30);"
          >
            Utstilling
          </a>
        </li>
        <li>
          <a
            class="nav_link"
            href="/favoritter"
            style="color: rgb(30, 30, 30);"
          >
            Favoritter
          </a>
        </li>
      </ul>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="landingpageText"
      >
        <h1>
          Velkommen til vektorgalleriet!
        </h1>
        <h2>
          Stedet der du kan tilpasse kunsten til humøret ditt!
        </h2>
        <a
          href="/utstilling"
        >
          Ta en titt på utstillingen!
        </a>
      </div>
      <div
        class="illustration"
      >
        <div
          class="glad"
        >
          <img
            alt=""
            class="landingpageIconsHappy"
            src="sun.png"
          />
          <img
            alt=""
            class="landingpageIconsHappy"
            src="emoji.png"
          />
        </div>
        <div
          class="trist"
        >
          <img
            alt=""
            class="landingpageIconsSad"
            src="water.png"
          />
          <img
            alt=""
            class="landingpageIconsSad"
            src="sad.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
